src/itemdrop.zig

```diff
@@ -674,46 +696,176 @@ pub const ItemDropRenderer = struct { // MARK: ItemDropRenderer
 				const light: [6]u8 = main.renderer.mesh_storage.getLight(blockPos[0], blockPos[1], blockPos[2]) orelse .{0} ** 6;
-				c.glUniform3fv(itemUniforms.ambientLight, 1, @ptrCast(&@max(
-					ambientLight*@as(Vec3f, @as(Vec3f, @floatFromInt(Vec3i{light[0], light[1], light[2]}))/@as(Vec3f, @splat(255))),
-					@as(Vec3f, @floatFromInt(Vec3i{light[3], light[4], light[5]}))/@as(Vec3f, @splat(255)),
-				)));
+				bindLightUniform(light, ambientLight);
 				pos -= playerPos;
 
 				const model = getModel(item);
 				c.glUniform1i(itemUniforms.modelIndex, model.index);
 				var vertices: u31 = 36;
 
 				var scale: f32 = 0.3;
+				var blockType: u16 = 0;
 				if(item == .baseItem and item.baseItem.block != null and item.baseItem.image.imageData.ptr == graphics.Image.defaultImage.imageData.ptr) {
-					const blockType = item.baseItem.block.?;
-					c.glUniform1i(itemUniforms.block, blockType);
+					blockType = item.baseItem.block.?;
 					vertices = model.len/2*6;
 				} else {
-					c.glUniform1i(itemUniforms.block, 0);
 					scale = 0.5;
 				}
+				bindModelUniforms(model.index, blockType);
 
 				var modelMatrix = Mat4f.translation(@floatCast(pos));
 				modelMatrix = modelMatrix.mul(Mat4f.rotationX(-rot[0]));
 				modelMatrix = modelMatrix.mul(Mat4f.rotationY(-rot[1]));
 				modelMatrix = modelMatrix.mul(Mat4f.rotationZ(-rot[2]));
 				modelMatrix = modelMatrix.mul(Mat4f.scale(@splat(scale)));
 				modelMatrix = modelMatrix.mul(Mat4f.translation(@splat(-0.5)));
-				c.glUniformMatrix4fv(itemUniforms.modelMatrix, 1, c.GL_TRUE, @ptrCast(&modelMatrix));
+				drawItem(vertices, modelMatrix);
+			}
+		}
+	}
 
-				c.glBindVertexArray(main.renderer.chunk_meshing.vao);
-				c.glDrawElements(c.GL_TRIANGLES, vertices, c.GL_UNSIGNED_INT, null);
+	inline fn getPos(x: u8, y: u8, z: u8) u32 {
+		return (z*4) + (y*2) + (x);
+	}
+
+	inline fn blendColors(a: [6]f64, b: [6]f64, t: f64) [6]f64 {
+		var result: [6]f64 = .{0, 0, 0, 0, 0, 0};
+		inline for(0..6) |i| {
+			result[i] = std.math.lerp(a[i], b[i], t);
+		}
+		return result;
+	}
+
+	pub fn renderDisplayItems(ambientLight: Vec3f, playerPos: Vec3d, time: u32) void {
+		if(!showItem) return;
+
+		const projMatrix: Mat4f = Mat4f.perspective(std.math.degreesToRadians(65), @as(f32, @floatFromInt(renderer.lastWidth))/@as(f32, @floatFromInt(renderer.lastHeight)), 0.05, 100);
+		const viewMatrix = Mat4f.identity();
+		bindCommonUniforms(projMatrix, viewMatrix, ambientLight, time);
+
+		const selectedItem = game.Player.inventory.getItem(game.Player.selectedSlot);
+		if(selectedItem) |item| {
+			var pos: Vec3d = Vec3d{0, 0, 0};
+			const rot: Vec3f = ItemDisplayManager.cameraFollow;
+			const blockPos: Vec3i = @intFromFloat(@floor(playerPos));
+			var localBlockPos = playerPos - @as(Vec3d, @floatFromInt(blockPos));
+			const quadrant: Vec3i = .{
+				if(localBlockPos[0] - 0.5 < 0) -1 else 1,
+				if(localBlockPos[1] - 0.5 < 0) -1 else 1,
+				if(localBlockPos[2] - 0.5 < 0) -1 else 1,
+			};
```

Comment:
If you subtract 0.5 from the player pos before flooring, then we can always use positive offsets here.